worlds/cv64/options.py

```diff
@@ -0,0 +1,490 @@
+from dataclasses import dataclass
+from Options import Choice, DefaultOnToggle, Range, Toggle, PerGameCommonOptions, StartInventoryPool
+
+
+class CharacterStages(Choice):
+    """Whether to include Reinhardt-only stages, Carrie-only stages, or both with or without branching paths at the end
+    of Villa and Castle Center."""
+    display_name = "Character Stages"
+    option_both = 0
+    option_branchless_both = 1
+    option_reinhardt_only = 2
+    option_carrie_only = 3
+    default = 0
+
+
+class StageShuffle(Toggle):
+    """Shuffles which stages appear in which stage slots. Villa and Castle Center will never appear in any character
+    stage slots if Character Stages is set to Both; they can only be somewhere on the main path.
+    Castle Keep will always be at the end of the line."""
+    display_name = "Stage Shuffle"
+
+
+class StartingStage(Choice):
+    """Which stage to start at if Stage Shuffle is turned on."""
+    display_name = "Starting Stage"
+    option_forest_of_silence = 0
+    option_castle_wall = 1
+    option_villa = 2
+    option_tunnel = 3
+    option_underground_waterway = 4
+    option_castle_center = 5
+    option_duel_tower = 6
+    option_tower_of_execution = 7
+    option_tower_of_science = 8
+    option_tower_of_sorcery = 9
+    option_room_of_clocks = 10
+    option_clock_tower = 11
+    default = "random"
+
+
+class WarpOrder(Choice):
+    """Arranges the warps in the warp menu in whichever stage order chosen,
+    thereby changing the order they are unlocked in."""
+    display_name = "Warp Order"
+    option_seed_stage_order = 0
+    option_vanilla_stage_order = 1
+    option_randomized_order = 2
+    default = 0
+
+
+class SubWeaponShuffle(Choice):
+    """Shuffles all sub-weapons in the game within each other in their own pool or in the main item pool."""
+    display_name = "Sub-weapon Shuffle"
+    option_off = 0
+    option_own_pool = 1
+    option_anywhere = 2
+    default = 0
+
+
+class SpareKeys(Choice):
+    """Puts an additional copy of every non-Special key item in the pool for every key item that there is.
+    Chance gives each key item a 50% chance of having a duplicate instead of guaranteeing one for all of them."""
+    display_name = "Spare Keys"
+    option_off = 0
+    option_on = 1
+    option_chance = 2
+    default = 0
+
+
+class HardItemPool(Toggle):
+    """Replaces some items in the item pool with less valuable ones, to make the item pool sort of resemble Hard Mode
+    in the PAL version."""
+    display_name = "Hard Item Pool"
+
+
+class Special1sPerWarp(Range):
+    """Sets how many Special1 jewels are needed per warp menu option unlock."""
+    range_start = 1
+    range_end = 10
+    default = 1
+    display_name = "Special1s Per Warp"
+
+
+class TotalSpecial1s(Range):
+    """Sets how many Speical1 jewels are in the pool in total.
+    If this is set to be less than Special1s Required x 7, it will increase by 1 until it isn't."""
```

Comment:
I don't believe this is true. I could be reading it wrong, but to me it looks like the code actually decrements s1's per warp instead of incrementing the total s1's

Reply:
Yeah, decrease is what I meant. Ooops.

Reply:
```suggestion
    If this is set to be less than Special1s Required x 7, it will decrease by 1 until it isn't."""
```

Reply:
That's not true though, because the thing that decreases is `special1s_per_warp`.

Reply:
I don't even have a setting called Special1s Required. XD
Fixing now.